docs/source/reference/tools/xtce_generator.rst

```diff
@@ -0,0 +1,81 @@
+.. _xtce_generator:
+
+Generating Telemetry XML with Python Script
+===========================================
+
+This document provides steps and information on how to use
+`xtce_generator_template.py` script as a base for users to generate
+telemetry XML files using the TelemetryGenerator class. The script is designed to
+simplify the process of creating telemetry definitions for various packet types.
+
+The script is located in the `tools/xtce_generation` directory. The script is called
+`xtce_generator_template.py`. The script is a ``template`` that can be modified to
+generate telemetry XML files for different packet types. Your new file should be
+called `xtce_generator_yourinstrument.py`.
+An example of how to use the script is `xtce_generator_codice.py` which is also
+located in the `tools/xtce_generation` directory.
+
+Prerequisites
+-------------
+
+Before you begin, ensure you have the following:
+
+- Python installed (version 3.9 or higher)
+- The required Python packages installed:
+    - pathlib
+    - tools.xtce_generation.telemetry_generator
```

Comment:
pathlib is a part of the standard library, and I don't think you need to install telemetry_generator but rather you just use it right?

I'd suggest indicating how you do this with the poetry environments and linking to those documents as well. Something similar to this, feel free to reword as you want.

> Generating XTCEs is only done whenever packet definitions get updated, and thus it is not a part of the main processing package. To use it there are a few extra dependencies like `pandas` that you can install with `poetry install --with-extras xtce`.

Reply:
I like this!